Documentation/gettingstarted/egress-gateway.rst

```diff
@@ -0,0 +1,172 @@
+
+**********************
+Egress Gateway (beta)
+**********************
+
+Egress gateway provides a way to configure the egress policy. Without
+this feature, pod traffic could only egress from the node it is running
+on and would be masqueraded with the node IP.
+
+This document explains how to configure Cilium to enable egress gateway
+as well as how to configure egress NAT policy to route and SNAT the egress
+traffic for a specific workload.
+
+.. include:: ../beta.rst
+
+Configure Cilium
+================
+
+.. include:: k8s-install-download-release.rst
+
+The Cilium Egress Gateway feature relies on :ref:`kubeproxy-free`,
+follow the guide to create a new deployment. The feature is disabled by default.
+Enable the feature by setting the ``localRedirectPolicy`` value to ``true``.
+
+.. parsed-literal::
+
+   helm install cilium |CHART_RELEASE| \\
+     --set enableEgressGateway=true
+
+Verify that Cilium agent pod is running.
+
+.. code-block:: shell-session
+
+    $ kubectl -n kube-system get pods -l k8s-app=cilium
+    NAME           READY   STATUS    RESTARTS   AGE
+    cilium-5ngzd   1/1     Running   0          3m19s
+
+Validate that the Cilium Egress NAT Policy CRD has been registered.
+
+.. code-block:: shell-session
+
+	   $ kubectl get crds
+	   NAME                              CREATED AT
+	   [...]
+	   ciliumegressnatpolicy.cilium.io   2021-03-26T02:11:21Z
+
+
+Optionally, Egress Gateway support can be enabled by setting
+``enable-egress-gateway`` to ``true`` in ConfigMap. By default, it is disabled.
+
+To set ``enable-egress-gateway`` to ``true``, it requires
+``enable-bpf-masquerade`` and ``kube-proxy-replacement``. It cannot be enabled
+with either of these 2 options set as ``false`` or ``disabled``.
+
+Architecture
+============
+
+Overall, some nodes in the cluster will be assigned as gateway nodes. These gateway nodes not
+only have a node IP, but also have additional pools of egress IPs attached to it. These egress IP
+addresses can be associated with certain workloads (pods) that run in the cluster. When certain
+packets are selected, instead of exiting the cluster immediately, they will be forwarded to a gateway
+node and SNAT-ed with an egress IP there. A tunnel interface will be created when Egress Gateway is
+enabled. The tunnel is for forwarding with Egress Gateway. All other traffic will keep their
+own behavior based on which tunnel mode is selected.
+
+The following 2 diagrams show the packet flow for both egress and return traffic when using direct
+routing mode. Note that the return packet is not going over a tunnel, while forwarding traffic is. 
+
+.. image:: images/cilium_egress_gateway_flow.png
+   :scale: 70%
+
+.. image:: images/cilium_egress_gateway_return_flow.png
+   :scale: 70%
+
+
+
+Create an External Service
+==========================
+
+Create an nginx service on a Linux node that is external to the existing Kubernetes
+cluster, and use it as the destination of the egress traffic.
```

Comment:
I understand this is useful to show that the received packets have been masqueraded to the expected egressIP, but most users testing this guide are likely to do so on managed Kubernetes platforms where reproducing this is not as straightforward.

Do we really need to show the source IP in the nginx logs? Could we instead show in some other way that the packets have gone through the egress gateway node as expected?

Reply:
I think this is just for an example of verification that is easy to understand and easy to use in practice.

Reply:
I agree it's a good example, but how are users with e.g. a GKE cluster supposed to follow this guide?

Reply:
They should be able to create a vm instance and install nginx. But you are right, this should be marked as optional. The user should be able to use their own external service to verify.

Reply:
I'm fine if we mark that step as optional.